### PR TITLE
feat(curriculum): added question on based form validation

### LINKED ISSUE #56210: [HACKTOBERFEST 2024] - Form Validation Quiz
> [!IMPORTANT]
> This issue is part of our 2024 Hacktoberfest initiative. If you are interested in working on this issue, refer to our [top-level tasks](https://github.com/freeCodeCamp/freeCodeCamp/issues/56170) to see if this issue is claimed, or to claim it if it is not.

This issue is to create the Form Validation quiz.

The file you should be working on is found in `/curriculum/challenges/english/25-front-end-development/quiz-form-validation`.

**All quiz questions need to relate to one of these topics:**

- Constraint Validation API
- `checkValidity()`
- e.preventDefault()
- submit event

> [!CAUTION]
> Questions which do not fall under these topics will be rejected.

You are welcome to include code blocks in your questions, but please keep them short (four lines or less).

Please do NOT open a pull request until you have completed all questions. Your quiz should have **exactly 20 questions**.

### PR DESCRIPTION
**Pull request by doosabhulaxmi**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #56210

### CODE REVIEW

File: curriculum/challenges/english/25-front-end-development/quiz-form-validation-with-javascript/66edd3403d7077eece6dc4b6.md [hunk 1]
```diff
@@ -17,439 +17,439 @@ Answer all of the questions below correctly to pass the quiz.
 
 #### --text--
 
-Placeholder question
+Which method is used to check if an input element's value satisfies the validation rules?
 
 #### --distractors--
 
-Placeholder distractor 1
+validateInput()
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-form-validation-with-javascript/66edd3403d7077eece6dc4b6.md:24]:
```suggestion
`validateInput()`
```

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
Before I can review this in depth, you are missing backticks around all of the methods/property names/references to code.

You will need to fix that. The review comment is an example of such a fix.

File: curriculum/challenges/english/25-front-end-development/quiz-form-validation-with-javascript/66edd3403d7077eece6dc4b6.md [hunk 2]
```diff
@@ -17,439 +17,439 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+The invalid field is highlighted and a message appears
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which actions will make `checkValidity()` run?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-form-validation-with-javascript/66edd3403d7077eece6dc4b6.md:196]:
```suggestion
Which action will make `checkValidity()` run?
```

File: curriculum/challenges/english/25-front-end-development/quiz-form-validation-with-javascript/66edd3403d7077eece6dc4b6.md [hunk 3]
```diff
@@ -17,439 +17,439 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+true
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+When does the `checkValidity()` method stop form submission?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-form-validation-with-javascript/66edd3403d7077eece6dc4b6.md:152]: I don't believe `checkValidity()` does this. It does fire an invalid event on the input elements, but NOT on the form itself (and the input-level event has no default behaviour).

File: curriculum/challenges/english/25-front-end-development/quiz-form-validation-with-javascript/66edd3403d7077eece6dc4b6.md [hunk 4]
```diff
@@ -17,439 +17,439 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 3
+To clear all input fields
 
 #### --answer--
 
-Placeholder answer
+To check if all required fields are filled and valid
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-form-validation-with-javascript/66edd3403d7077eece6dc4b6.md:234]:
```suggestion
To check if all inputs meet their specified constraints
```

File: curriculum/challenges/english/25-front-end-development/quiz-form-validation-with-javascript/66edd3403d7077eece6dc4b6.md [hunk 5]
```diff
@@ -17,439 +17,439 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+The form submits and refreshes the page
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which actions can be prevented using `e.preventDefault()`?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-form-validation-with-javascript/66edd3403d7077eece6dc4b6.md:328]: Hmm, this one's a little off. Technically `e.preventDefault()` doesn't prevent *submitting* a form, it prevents the default behaviour *caused* by submitting a form (re: the page refresh and POST call).

File: curriculum/challenges/english/25-front-end-development/quiz-form-validation-with-javascript/66edd3403d7077eece6dc4b6.md [hunk 6]
```diff
@@ -17,439 +17,439 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+It sends form data to the server specified in the form's action
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which method should you use to handle form data on the submit event in JavaScript?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-form-validation-with-javascript/66edd3403d7077eece6dc4b6.md:416]: Hmm. Technically, these are all the same method, with a different argument.

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]

**Comment by doosabhulaxmi** [comment]
@naomi-lgbt changes done on form validation.
please check now

File: curriculum/challenges/english/25-front-end-development/quiz-form-validation-with-javascript/66edd3403d7077eece6dc4b6.md [hunk 7]
```diff
@@ -17,439 +17,439 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --text--
 
-Placeholder question
+Which event should you listen for to handle form data when the form is submitted in JavaScript?
 
 #### --distractors--
 
-Placeholder distractor 1
+input
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-form-validation-with-javascript/66edd3403d7077eece6dc4b6.md:420]:
```suggestion
`"input"`
```

File: curriculum/challenges/english/25-front-end-development/quiz-form-validation-with-javascript/66edd3403d7077eece6dc4b6.md [hunk 8]
```diff
@@ -17,439 +17,439 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 2
+change
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-form-validation-with-javascript/66edd3403d7077eece6dc4b6.md:424]:
```suggestion
`"change"`
```

File: curriculum/challenges/english/25-front-end-development/quiz-form-validation-with-javascript/66edd3403d7077eece6dc4b6.md [hunk 9]
```diff
@@ -17,439 +17,439 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 3
+focus
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-form-validation-with-javascript/66edd3403d7077eece6dc4b6.md:428]:
```suggestion
`"focus"`
```

File: curriculum/challenges/english/25-front-end-development/quiz-form-validation-with-javascript/66edd3403d7077eece6dc4b6.md [hunk 10]
```diff
@@ -17,439 +17,439 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --answer--
 
-Placeholder answer
+submit
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-form-validation-with-javascript/66edd3403d7077eece6dc4b6.md:432]:
```suggestion
`"submit"`
```

File: curriculum/challenges/english/25-front-end-development/quiz-form-validation-with-javascript/66edd3403d7077eece6dc4b6.md [hunk 11]
```diff
@@ -17,439 +17,439 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 2
+Stop the form from being displayed
 
 ---
 
-Placeholder distractor 3
+Set the form's method to GET
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-form-validation-with-javascript/66edd3403d7077eece6dc4b6.md:450]:
```suggestion
Set the form's `method` to `"GET"`
```

**Review by majestic-owl448 — COMMENTED** [review]
some stylistic things

File: curriculum/challenges/english/25-front-end-development/quiz-form-validation-with-javascript/66edd3403d7077eece6dc4b6.md [hunk 12]
```diff
@@ -17,439 +17,439 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 3
+`customMessage`
 
 #### --answer--
 
-Placeholder answer
+`setCustomValidity()`
```
[the remark this answers is not in the document]
Comment (doosabhulaxmi) [reply curriculum/challenges/english/25-front-end-development/quiz-form-validation-with-javascript/66edd3403d7077eece6dc4b6.md:80]: method

**Comment by doosabhulaxmi** [comment]
@ilenia-magoni  changes done.
please check now

File: curriculum/challenges/english/25-front-end-development/quiz-form-validation-with-javascript/66edd3403d7077eece6dc4b6.md [hunk 13]
```diff
@@ -17,439 +17,439 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+The browser displays a validation message
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which property is used to customize the message shown during validation failure?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-form-validation-with-javascript/66edd3403d7077eece6dc4b6.md:64]:
```suggestion
Which method is used to customize the message shown during validation failure?
```

File: curriculum/challenges/english/25-front-end-development/quiz-form-validation-with-javascript/66edd3403d7077eece6dc4b6.md [hunk 14]
```diff
@@ -17,439 +17,439 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --distractors--
 
-Placeholder distractor 1
+`setErrorMessage`
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-form-validation-with-javascript/66edd3403d7077eece6dc4b6.md:68]:
```suggestion
`setErrorMessage()`
```

File: curriculum/challenges/english/25-front-end-development/quiz-form-validation-with-javascript/66edd3403d7077eece6dc4b6.md [hunk 15]
```diff
@@ -17,439 +17,439 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 2
+`setValidationMessage`
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-form-validation-with-javascript/66edd3403d7077eece6dc4b6.md:72]:
```suggestion
`setValidationMessage()`
```

File: curriculum/challenges/english/25-front-end-development/quiz-form-validation-with-javascript/66edd3403d7077eece6dc4b6.md [hunk 16]
```diff
@@ -17,439 +17,439 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 3
+`customMessage`
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-form-validation-with-javascript/66edd3403d7077eece6dc4b6.md:76]:
```suggestion
`customMessage()`
```

File: curriculum/challenges/english/25-front-end-development/quiz-form-validation-with-javascript/66edd3403d7077eece6dc4b6.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (naomi-lgbt) [reply curriculum/challenges/english/25-front-end-development/quiz-form-validation-with-javascript/66edd3403d7077eece6dc4b6.md:80]: Then the question and distractors should reflect that.

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]

**Comment by doosabhulaxmi** [comment]
@naomi-lgbt changes done please check now.

**Review by naomi-lgbt — APPROVED** [review]

File: curriculum/challenges/english/25-front-end-development/quiz-form-validation-with-javascript/66edd3403d7077eece6dc4b6.md (hunk 16 above)
Comment (doosabhulaxmi) [reply curriculum/challenges/english/25-front-end-development/quiz-form-validation-with-javascript/66edd3403d7077eece6dc4b6.md]: changes done @naomi-lgbt

**Comment by doosabhulaxmi** [comment]
@ilenia-magoni please check

**Comment by doosabhulaxmi** [comment]
please review

**Comment by ilenia-magoni** [comment]
@doosabhulaxmi please stop pinging maintainers

**Review by Dario-DC — APPROVED** [review]

**Pull request merged**